### PR TITLE
Function replacement implementation v1

### PR DESCRIPTION
**Pull request by zheng-ningxin**
In this PR, we implement the framework of the function replacement in compression and the replace function for the view function.

### CODE REVIEW

**Comment by liuzhe-lz** [comment]
Deferred to next release.

File: nni/compression/pytorch/speedup/compressor.py [hunk 1]
```diff
@@ -378,6 +389,48 @@ def replace_compressed_modules(self):
         with torch.no_grad():
             for unique_name in self.auto_inferences:
                 self.replace_submodule(unique_name)
+            # the replaced model training state is set to True by default
+            self.bound_model.train(False)
```
[the remark this answers is not in the document]
Comment (zheng-ningxin) [reply nni/compression/pytorch/speedup/compressor.py:399]: Yes, we already restore the original training state at the end of function `speedup_model`

File: nni/compression/pytorch/speedup/compress_modules.py [hunk 2]
```diff
@@ -531,13 +564,90 @@ def replace_layernorm(layernorm, masks):
...
+    # max_norm = embedding.max_norm
+    # norm_type = embedding.norm_type
+    # scale_grad_by_freq = embedding.scale_grad_by_freq
+    # new_embedding = torch.nn.Embedding(num_embeddings, n_remain_dim, padding_idx=padding_idx,
+    #                                    max_norm=max_norm, norm_type=norm_type, scale_grad_by_freq=scale_grad_by_freq)
+    # new_embedding.weight.data = torch.index_select(embedding.weight.data, 1, remained_dim)
+    # return new_embedding
+
+
+def replace_view(trace_model, cpp_node, masks):
```
Comment (J-shang) [inline nni/compression/pytorch/speedup/compress_modules.py:600]: Do we already fix the view issue in our last discussion?

Comment (zheng-ningxin) [reply nni/compression/pytorch/speedup/compress_modules.py:600]: Nice catch, I forgot this one. Thanks.

**Review by J-shang — APPROVED** [review]
Better to add a warning if replacing the view, because output may not align after replacement

**Pull request closed without merging**